Attach `dd.trace_id` to json formatted log messages (#4764)
apollographql/router: docs/source/configuration/telemetry/exporters/tracing/datadog.mdx

For more details about Datadog configuration, see [Datadog Agent configuration](https://docs.datadoghq.com/opentelemetry/otlp_ingest_in_the_agent/?tab=host).

### Enabling log correlation

> To enable Datadog log correlation `dd.trace_id` must be configured to appear on the `router` span:

Reviewer: ```suggestion
To enable Datadog log correlation, you must configure `dd.trace_id` to appear on the `router` span:
```

  
```yaml title="router.yaml"
telemetry:
  instrumentation:
    spans:
      mode: spec_compliant
      router:
        attributes:

>           dd.trace_id: true

Reviewer: ```suggestion
          dd.trace_id: true #highlight-line
```

```

> Your `json` formatted log messages will automatically output `dd.trace_id` on each log message if `dd.trace_id` was detected on the `router` span.

Reviewer: ```suggestion
Your JSON formatted log messages will automatically output `dd.trace_id` on each log message if `dd.trace_id` was detected on the `router` span.
```
